### ISSUE TITLE
Makefile in master removes possible current release tarball

### ISSUE THREAD

**Issue by levitte** [openssl/openssl#14981]
I was surprised, today, to see this:

``` console
$ make clean
...
rm -f ../openssl-3.0.0-alpha15-dev.tar
```

https://github.com/openssl/openssl/blob/cd28d129b6a5b84ac40b4a3f8060a6f764aa02b4/Configurations/unix-Makefile.tmpl#L564

This is pretty old...  it looks like it appeared in 567a9e6fe0ff3badfa22cf018d87c94ed5a8aeb3, i.e. as part of the initial unified build effort.  I simply hadn't noticed.

I would say that we should not do this, especially since we might remove the source tarball that is the origin of the source tree, i.e. force our users to re-download it after a `make clean` if they end up wanting to restart with a pristine source tree.

**Comment by t8m** [comment]
This removes the ungzipped tarfile and not the gzipped one. So I do not think this is that big deal. But probably the creation of the gzipped release tarball should create the tar.gz straight away and avoid the intermediary .tar step.

**Comment by levitte** [comment]
@t8m, you might want to look at `util/mktar.sh`, it doesn't write any intermediary uncompressed tarball...  it's possible it once did, but even then, we really have no business removing files that are outside the source tree.

**Comment by mspncp** [comment]
_m**rde. @levitte beat me again by a few seconds..._
```
util/mktar.sh:git archive --worktree-attributes -9 --prefix="$NAME/" -o $TARFILE.gz -v HEAD
```

**Comment by t8m** [comment]
Yeah, I was looking at the unix-Makefile.tmpl as of 567a9e6
So yes, there is really no reason to rm the $TARFILE anymore.

**Comment by levitte** [comment]
Side note: @mspncp, "m*rde" is spelt with one "*", not two :wink:

**Comment by richsalz** [comment]
I think it's surprising when "make clean" touches stuff above the directory, i.e., ".."

**Comment by levitte** [comment]
> I think it's surprising when "make clean" touches stuff above the directory, i.e., ".."

Exactly

**Comment by levitte** [comment]
Er, this was fixed with #14985.  Closing
